jetty-core/jetty-compression/jetty-compression-server/src/main/java/org/eclipse/jetty/compression/server/CompressionConfig.java

```diff
@@ -0,0 +1,692 @@
+{
+    private static final Logger LOG = LoggerFactory.getLogger(CompressionConfig.class);
+
+    private static final int MIN_FUNCTIONAL_SIZE = 23;
+    private static final int DEFAULT_MIN_COMPRESS_SIZE = 32;
+    /**
+     * Set of `Content-Encoding` encodings that are supported by this configuration.
+     */
+    private final IncludeExclude<String> decompressEncodings = new IncludeExclude<>();
+    /**
+     * Set of `Accept-Encoding` encodings that are supported by this configuration.
+     */
+    private final IncludeExclude<String> compressEncodings = new IncludeExclude<>();
+    /**
+     * Set of HTTP Methods that are supported by this configuration.
+     */
+    private final IncludeExclude<String> methods = new IncludeExclude<>();
+    // TODO: rename or eliminate
+    private final IncludeExclude<String> inflatePaths = new IncludeExclude<>(PathSpecSet.class);
+    // TODO: rename or eliminate
+    private final IncludeExclude<String> paths = new IncludeExclude<>(PathSpecSet.class);
+    private final IncludeExclude<String> mimeTypes = new IncludeExclude<>(AsciiLowerCaseSet.class);
+    private int compressSizeMinimum = DEFAULT_MIN_COMPRESS_SIZE;
+    private int decompressBufferSize = -1;
+    private boolean syncFlush = false;
+    private HttpField vary = new PreEncodedHttpField(HttpHeader.VARY, HttpHeader.ACCEPT_ENCODING.asString());
+
+    /**
+     * Adds excluded Path Specs for request filtering on request inflation.
+     *
+     * <p>
+     * There are 2 syntaxes supported, Servlet <code>url-pattern</code> based, and
+     * Regex based.  This means that the initial characters on the path spec
+     * line are very strict, and determine the behavior of the path matching.
+     * <ul>
+     * <li>If the spec starts with <code>'^'</code> the spec is assumed to be
+     * a regex based path spec and will match with normal Java regex rules.</li>
+     * <li>If the spec starts with <code>'/'</code> then spec is assumed to be
+     * a Servlet url-pattern rules path spec for either an exact match
+     * or prefix based match.</li>
+     * <li>If the spec starts with <code>'*.'</code> then spec is assumed to be
+     * a Servlet url-pattern rules path spec for a suffix based match.</li>
+     * <li>All other syntaxes are unsupported</li>
+     * </ul>
+     * <p>
+     * Note: inclusion takes precedence over exclude.
+     *
+     * @param pathspecs Path specs (as per servlet spec) to exclude. If a
+     * ServletContext is available, the paths are relative to the context path,
+     * otherwise they are absolute.<br>
+     * For backward compatibility the pathspecs may be comma separated strings, but this
+     * will not be supported in future versions.
+     * @see #addIncludedInflationPaths(String...)
+     * @deprecated review "backward compatibility" here.
+     */
+    @Deprecated // TODO review "backward compatibility" here. (use of csvSplit with var-args)
+    public void addExcludedInflationPaths(String... pathspecs)
```

Comment:
It might be simpler to have `CompressionConfig` as an immutable class and then some static methods for combining them.